File: our-knowledge-front/src/backend/filterService.js
import {appFetch, config} from "./appFetch";

export const findFilters = (onSuccess) =>
    appFetch(`/filter/list`,
        config('POST', {}), onSuccess);

export const findByFilterId = (filterId, onSuccess) =>
    appFetch(`/filter`,
        config('Post', {filterId}), onSuccess);

export const getDefaultFilter = (onSuccess) =>
    appFetch(`/filter/default`,
        config('Post', {}), onSuccess);

export const saveFilter = (filterName, onSuccess) =>
    appFetch(`/filter/save`,
        config('Post', {filterName}), onSuccess);

export const clearFilter = (onSuccess) =>
    appFetch(`/filter/clear`,
        config('Post', {}), onSuccess);

export const deleteFilter = (filterId, onSuccess) =>
    appFetch(`/filter/delete`,
        config('Post', {filterId}), onSuccess);

export const updateFilterParam = (filterParamId, filterId, technologyId, mandatory, recommended, onSuccess) =>
    appFetch(`/filter/filterParam/update`,
        config('Post', {filterParamId, filterId, technologyId, mandatory, recommended}),
        onSuccess);

export const useProjectAsFilter = (projectId, onSuccess) =>
    appFetch(`/filter/createByProject`,
        config('Post', {projectId}),
        onSuccess);

export const useUserAsFilter = (userId, onSuccess) =>
    appFetch(`/filter/createByUser`,
        config('Post', {userId}),
        onSuccess);

File: OurKnowledgeBackend/src/main/java/com/example/ourknowledgebackend/service/FilterService.java
package com.example.ourknowledgebackend.service;

import com.example.ourknowledgebackend.exceptions.DuplicateInstanceException;
import com.example.ourknowledgebackend.exceptions.InstanceNotFoundException;
import com.example.ourknowledgebackend.exceptions.PermissionException;
import com.example.ourknowledgebackend.model.CompleteFilter;
import com.example.ourknowledgebackend.model.entities.Filter;

import javax.naming.directory.InvalidAttributesException;
import java.util.List;

public interface FilterService {

    List<Filter> listFilter(Long userId) throws InstanceNotFoundException;

    CompleteFilter getFilter(Long filterId) throws InstanceNotFoundException;

    CompleteFilter getDefaultFilter(Long userId) throws InstanceNotFoundException;

    void saveFilter(Long userId, String name) throws InstanceNotFoundException, DuplicateInstanceException;

    void clearFilter(Long userId) throws InstanceNotFoundException;

    void deleteFilter(Long userId, Long filterId) throws InstanceNotFoundException, PermissionException;

    void createByProject(Long userId, Long projectId) throws InstanceNotFoundException;

    Long updateFilterParam(Long userId, Long filterParamId, Long filterId, Long technologyId, Boolean mandatory, Boolean recommended) throws InstanceNotFoundException, InvalidAttributesException, PermissionException, DuplicateInstanceException;

}
